EIPS/eip-4834.md

```diff
@@ -0,0 +1,224 @@
+    ///             `getDomain(name).canDeleteSubdomain(msg.sender, this, subdomain)` is `false`.
+    /// @param      updater The account that may or may not be able to delete a subdomain
+    /// @param      name The subdomain to delete
+    /// @return     Whether an account can delete the subdomain
+    function canDeleteDomain(address updater, string memory name) external view returns (bool);
+
+
+    //// Subdomain Access Control
+
+    /// @notice     Get if an account can point this domain as a subdomain
+    /// @dev        May be called by `canSetDomain` of the parent domain - implement access control here!!!
+    /// @param      updater The account that may be moving the subdomain
+    /// @param      name The subdomain name
+    /// @param      parent The parent domain
+    /// @return     Whether an account can update the subdomain
+    function canPointSubdomain(address updater, string memory name, IDomain parent) external view returns (bool);
+
+    /// @notice     Get if an account can move the subdomain away from the current domain
+    /// @dev        May be called by `canSetDomain` of the parent domain - implement access control here!!!
+    /// @param      updater The account that may be moving the subdomain
+    /// @param      name The subdomain name
+    /// @param      parent The parent domain
+    /// @param      newSubdomain The domain that will be set next
+    /// @return     Whether an account can update the subdomain
+    function canMoveSubdomain(address updater, string memory name, IDomain parent, address newSubdomain) external view returns (bool);
+
+    /// @notice     Get if an account can point this domain as a subdomain
+    /// @dev        May be called by `canDeleteDomain` of the parent domain - implement access control here!!!
+    /// @param      updater The account that may or may not be able to delete a subdomain
+    /// @param      name The subdomain to delete
+    /// @param      parent The parent domain
+    /// @return     Whether an account can delete the subdomain
+    function canDeleteSubdomain(address updater, string memory name, IDomain parent) external view returns (bool);
+}
+```
+
+### Name Resolution
+To resolve a name (like `"a.b.c"`), you first split it by the delimiter (resulting in something like `["a", "b", "c"]`). Set `domain` initially to the root domain.
+
+Pop off the last element of the array (`"c"`), then call `domain.hasDomain(lastElement)`. If it's `false`, then the domain resolution fails. Otherwise, set the domain to `domain.getDomain(lastElement)`. Repeat until there are no more lists.
+
+Here is a solidity function that resolves a name:
+
+```sol
+function resolve(string[] calldata splitName, IDomain root) public view returns (address) {
+    IDomain current = root;
+    for (uint i = splitName.length - 1; i >= 0; i--) {
+        // Ensure that domain is actually a domain
+        require(current.supportsInterface(type(IDomain).interfaceId));
```

Comment:
I think it's important to mention this check (`supportsInterface`) as part of the algorithm described above.

Reply:
The check was just because I wasn't sure what would happen. I'll remove it since I'm pretty sure it won't break.